docs/proposed/config-split.md

```diff
@@ -0,0 +1,450 @@
+Configuration file split
+========================
+
+## Why splitting the configuration file?
+
+The objective of the repository split is to reach the point where in Mbed TLS
+all the cryptography code and its tests are located in a tf-psa-crypto
+directory that just contains the TF-PSA-Crypto repository as a submodule.
+The cryptography APIs exposed by Mbed TLS are just the TF-PSA-Crypto ones.
+Mbed TLS relies solely on the TF-PSA-Crypto build system to build its
+cryptography library and its tests.
+
+The TF-PSA-Crypto configuration file `tf_psa_crypto_config.h` configures
+entirely the cryptography interface exposed by Mbed TLS through TF-PSA-Crypto.
+Mbed TLS configuration is splitted in two files: `mbedtls_config.h` for TLS and
+x509, `tf_psa_crypto_config.h` for the cryptography.
+
+## How do we split the configuration file?
+
+We extend the so called PSA cryptographic configuration scheme based on
+`mbedtls_config.h` and `crypto_config.h`. The configuration file `crypto_config.h`
+is extended to become the TF-PSA-Crypto configuration file, `mbedtls_config.h`
+becomes the configuration file for the TLS and x509 libraries. All the options
+to select the cryptographic mechanisms and to configure their implementation
+are moved from `mbedtls_config.h` to `(tf_psa_)crypto_config.h`.
+
+The configuration options that are relevant to both Mbed TLS and TF-PSA-Crypto
+like platform or system ones are moved to `(tf_psa_)crypto_config.h`. That way
+they are available in both repositories (as Mbed TLS includes
+`tf_psa_crypto_config.h`) without duplication. Later, we may duplicate or
+create aliases for some of them to align with the naming conventions of the
+repositories.
+
+The layout of options into sections in `mbedtls_config.h` does not suit
```

Comment:
It's not very clear what you mean: that `tf_psa_crypto_config.h` can't have the same layout as the 4.x `mbedtls_config.h` that you describe below, or that `tf_psa_crypto_config.h` can't have the same layout as the 3.x `mbedtls_config.h`? I read your sentence as the former, but in context it seems that you're arguing the latter (we can't keep the existing layout, hence we've designed a new layout).

Apart from sections that are intrinsically specific to one project (e.g. drivers in TF-PSA-Crypto, X.509 vs TLS in Mbed TLS), I don't see why the layouts would need to differ.

Reply:
I've changed that completely.